go/consensus/tendermint/tendermint.go

```diff
@@ -835,19 +835,36 @@ func (t *tendermintService) GetHeight(ctx context.Context) (int64, error) {
 	return blk.Header.Height, nil
 }
 
-func (t *tendermintService) GetBlockResults(height *int64) (*tmrpctypes.ResultBlockResults, error) {
+func (t *tendermintService) GetBlockResults(height int64) (*tmrpctypes.ResultBlockResults, error) {
 	if t.client == nil {
 		panic("client not available yet")
 	}
 
-	result, err := t.client.BlockResults(height)
+	// As in GetTendermintBlock above, get the latest tendermint block height
+	// from our mux.
+	var tmHeight int64
+	if height == consensusAPI.HeightLatest {
+		tmHeight = t.mux.BlockHeight()
+		if tmHeight == 0 {
+			// No committed blocks yet.
+			return nil, nil
+		}
+	} else {
+		tmHeight = height
+	}
+
+	result, err := t.client.BlockResults(&tmHeight)
 	if err != nil {
 		return nil, fmt.Errorf("tendermint: block results query failed: %w", err)
 	}
 
 	return result, nil
 }
 
+func (t *tendermintService) GetEvents(ctx context.Context, height int64) ([]roothashAPI.Event, error) {
```

Comment:
Why is this here? It gets you roothash events?

Oh I see, you added it here because the roothash service is not exposed directly. This doesn't seem right. If anywhere this should be added to the `RuntimeClient` API instead. But seeing what this contains I would be ok with just not adding it (and no need to test it in txsource for now as long as there is a unit test).

Reply:
Yes, there is no RoothashClient.  I've removed this code now.